SystemTextJsonPatch/Internal/Proxies/DictionaryTypedPropertyProxy.cs

```diff
@@ -3,12 +3,12 @@
 
 namespace SystemTextJsonPatch.Internal.Proxies
 {
-	internal sealed class DictionaryTypedPropertyProxy : IPropertyProxy
+	internal sealed class DictionaryTypedPropertyProxy<TKey, TValue> : IPropertyProxy
 	{
-		private readonly IDictionary<string, object?> _dictionary;
-		private readonly string _propertyName;
+		private readonly IDictionary<TKey, TValue?> _dictionary;
+		private readonly TKey _propertyName;
 
-		internal DictionaryTypedPropertyProxy(IDictionary<string, object?> dictionary, string propertyName)
+		public DictionaryTypedPropertyProxy(IDictionary<TKey, TValue?> dictionary, TKey propertyName)
```

Comment:
Does the constructor really have to be public?

Reply:
Year, for `Activator.CreateInstance()`. The type itself is `internal`. I can adjust the call to `Activator.CreateInstance()` with flags to create using an internal ctor if you want?

Reply:
yeah well I was trying to hide these from public as the implementation details might change